### PR TITLE
[AIR] Improve `MaxAbsScaler` docstring

### PR DESCRIPTION
**Pull request by bveeramani**
## Why are these changes needed?

These changes are needed to make `MaxAbsScaler` more understandable.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [x] This PR is not tested :(

### CODE REVIEW

File: python/ray/data/preprocessors/scaler.py [hunk 1]
```diff
@@ -99,16 +99,53 @@ def __repr__(self):
...
-    range [-1, 1].
+    .. math::
+
+        x' = \frac{x}{\max{\vert x \vert}}
+
+    where :math:`x` is the column and :math:`x'` is the transformed column. If
+    :math:`\max{\vert x \vert} = 0` (i.e., the column contains all zeros), then the
+    column is unmodified.
+
+    Transformed values are always in the range :math:`[-1, 1]`.
```
Comment (matthewdeng) [inline python/ray/data/preprocessors/scaler.py:114]: Ah this is only true for the _fitted_ dataset. If you transform another dataset which has larger values, the values will be outside this range.

File: python/ray/data/preprocessors/scaler.py [hunk 2]
```diff
@@ -99,16 +99,53 @@ def __repr__(self):
...
+
+    .. note::
+        This is recommend way to scale sparse data.
```
Comment (matthewdeng) [inline python/ray/data/preprocessors/scaler.py:117]:
```suggestion
        This is the recommended way to scale sparse data.
```

Comment (matthewdeng) [reply python/ray/data/preprocessors/scaler.py:117]: What are your thoughts on linking other scalers here? My natural reaction to reading this comment was asking "what if my data is not sparse?"

Alternatively we can follow up with a user guide.

File: python/ray/data/preprocessors/scaler.py [hunk 3]
```diff
@@ -99,16 +99,53 @@ def __repr__(self):
 
 
 class MaxAbsScaler(Preprocessor):
-    """Scale values within columns based on the absolute max value.
+    r"""Scale each column by its absolute max value.
```
Comment (matthewdeng) [inline python/ray/data/preprocessors/scaler.py:102]: Thinking through this, maybe we want to establish a pattern in the docstrings to describe what fit and transform individually do.

File: python/ray/data/preprocessors/scaler.py (hunk 2 above)
Comment (bveeramani) [reply python/ray/data/preprocessors/scaler.py]: Fixed

File: python/ray/data/preprocessors/scaler.py (hunk 1 above)
Comment (bveeramani) [reply python/ray/data/preprocessors/scaler.py:114]: Fixed

**Comment by bveeramani** [comment]
Closing in favor of #27215

**Pull request closed without merging**